plum/function.py

```diff
@@ -417,6 +417,8 @@ def invoke(self, *types: TypeHint) -> Callable:
         def wrapped_method(*args, **kw_args):
             return _convert(method(*args, **kw_args), return_type)
 
+        wrapped_method.__wrapped_by_plum__ = method
```

Comment:
Can you add a docstring to explain what this does and why?

Reply:
Definitely! Will also add a test which illustrates the edge case. Sorry, haven’t had a moment yet to do this.

Reply:
Added in the tests!